helm-v3/005-test.md

```diff
@@ -0,0 +1,45 @@
+# Test Command
+
+## Overview
+
+`helm test` gives a chart author the ability to define actions on a helm release which validate that the chart is running as intended. This is useful for ensuring that a `helm install` or `helm upgrade` created/modified Kubernetes resources successfully and that the consequential `helm release` is functional.
+* A **test** is defined as a command and some arguments executed from a container running in a Kubernetes Pod. If the command completes inside the container with an exit code of 0, it is successful. All other exit codes indicate the test failed.
+* Tests must be defined in the `templates/tests/` directory. This used to be recommended as a best practice. See motivations for change below:
```

Comment:
I'm not a fan of requiring test resources to be in a separate directory for a few reasons:

1. It's already possible to organize tests in this way if we use annotations
1. An annotation is still required in this proposal, and getting both the annotation and directory placement right could be a point of confusion for users
1. Resources currently defined within `templates/tests` would no longer be rendered with the chart if we say they'll be automatically interpreted as supporting resources for tests (another breaking change)
1. Sharing templates between helm tests and hooks might become impossible or require helper templates (eww)

Reply:
I agree. We shouldn't require a specific directory. An annotation should suffice.